File: services/customer-service/src/main/java/com/ev/customer_service/service/CartService.java
package com.ev.customer_service.service;

import com.ev.customer_service.dto.request.AddToCartRequest;
import com.ev.customer_service.dto.request.UpdateCartItemRequest;
import com.ev.customer_service.dto.response.CartItemResponse;
import com.ev.customer_service.dto.response.CartSummaryResponse;

import java.util.List;

public interface CartService {

    /**
     * Thêm sản phẩm vào giỏ hàng
     * Nếu sản phẩm đã tồn tại, tăng quantity
     */
    CartItemResponse addToCart(Long customerId, AddToCartRequest request);

    /**
     * Lấy tất cả items trong giỏ hàng của customer
     */
    List<CartItemResponse> getCartItems(Long customerId);

    /**
     * Lấy tổng quan giỏ hàng (items + tổng tiền)
     */
    CartSummaryResponse getCartSummary(Long customerId);

    /**
     * Cập nhật cart item (quantity, selected features)
     */
    CartItemResponse updateCartItem(Long customerId, Long cartItemId, UpdateCartItemRequest request);

    /**
     * Xóa một item khỏi giỏ hàng
     */
    void removeCartItem(Long customerId, Long cartItemId);

    /**
     * Xóa tất cả items khỏi giỏ hàng
     */
    void clearCart(Long customerId);

    /**
     * Đếm số lượng items trong giỏ hàng
     */
    Long getCartItemCount(Long customerId);

    /**
     * Đếm số lượng items trong giỏ hàng bằng profileId
     */
    Long getCartItemCountByProfileId(String profileId);
}

File: frontend/customer-app/src/services/cartService.js
import api from "./api.js";

/**
 * Cart Service
 * Provides API methods for shopping cart operations
 */

// Add item to cart
export const addToCart = (customerId, cartData) =>
  api.post(`/cart/${customerId}`, cartData).then((res) => res.data);

// Get all cart items
export const getCartItems = (customerId) =>
  api.get(`/cart/${customerId}/items`).then((res) => res.data);

// Get cart summary (items + total)
export const getCartSummary = (customerId) =>
  api.get(`/cart/${customerId}/summary`).then((res) => res.data);

// Update cart item
export const updateCartItem = (customerId, cartItemId, updateData) =>
  api.put(`/cart/${customerId}/items/${cartItemId}`, updateData).then((res) => res.data);

// Remove item from cart
export const removeCartItem = (customerId, cartItemId) =>
  api.delete(`/cart/${customerId}/items/${cartItemId}`).then((res) => res.data);

// Clear entire cart
export const clearCart = (customerId) =>
  api.delete(`/cart/${customerId}/clear`).then((res) => res.data);

// Get cart item count
export const getCartItemCount = (customerId) =>
  api.get(`/cart/${customerId}/count`).then((res) => res.data);
